libcloud/common/durabledns.py

```diff
@@ -0,0 +1,294 @@
+                    extra['retry'] = int(child.text.strip())
+                elif child.tag == 'expire':
+                    extra['expire'] = int(child.text.strip())
+                elif child.tag == 'minimum':
+                    extra['minimum'] = int(child.text.strip())
+                else:
+                    if child.text:
+                        extra[child.tag] = child.text.strip()
+                    else:
+                        extra[child.tag] = ''
+                    zone_dict['extra'] = extra
+            objects.append(zone_dict)
+        # parse response from getRecordResponse
+        if 'getRecordResponse' in method_resp.tag:
+            answer = method_resp.getchildren()[0]
+            for child in method_resp.getchildren():
+                if child.tag == 'id' and child.text:
+                    record_dict['id'] = child.text.strip()
+                elif child.tag == 'name' and child.text:
+                    record_dict['name'] = child.text.strip()
+                elif child.tag == 'type' and child.text:
+                    record_dict['type'] = child.text.strip()
+                elif child.tag == 'data' and child.text:
+                    record_dict['data'] = child.text.strip()
+                elif child.tag == 'aux' and child.text:
+                    record_dict['aux'] = child.text.strip()
+                elif child.tag == 'ttl' and child.text:
+                    record_dict['ttl'] = child.text.strip()
+            if not record_dict:
+                error_dict['ERRORMESSAGE'] = 'Record does not exist'
+                error_dict['ERRORCODE'] = 404
+                errors.append(error_dict)
+            objects.append(record_dict)
+            record_dict = {}
+        if 'createZoneResponse' in method_resp.tag:
+            answer = method_resp.getchildren()[0]
+            if answer.tag == 'return' and answer.text:
+                record_dict['id'] = answer.text.strip()
+            objects.append(record_dict)
+        # catch Record does not exists error when deleting record
+        if 'deleteRecordResponse' in method_resp.tag:
+            answer = method_resp.getchildren()[0]
+            if 'Record does not exists' in answer.text.strip():
+                errors.append({'ERRORMESSAGE': answer.text.strip(),
+                               'ERRORCODE': self.status})
+        # parse response in createRecordResponse
+        if 'createRecordResponse' in method_resp.tag:
+            answer = method_resp.getchildren()[0]
+            record_dict['id'] = answer.text.strip()
+            objects.append(record_dict)
+            record_dict = {}
+
+        return (objects, errors)
+
+    def parse_body(self):
+        # A problem arise in the api response because there are undeclared
+        # xml namespaces. In order to fix that at the moment, we use the
+        # _fix_response method to clean up since we won't always have lxml
+        # library.
+        self._fix_response()
```

Comment:
I know you mentioned to try to use most of i can parent methods. But this api returns an xml with xml namespaces empty and ElementTree don't like that. I had to make this "_fix_response" method to clean those tags.